synapse/handlers/room.py

```diff
@@ -367,11 +368,14 @@ def clone_existing_room(
         # Raise the requester's power level in the new room if necessary
         current_power_level = power_levels["users"][requester.user.to_string()]
         if current_power_level < needed_power_level:
-            # Assign this power level to the requester
-            power_levels["users"][requester.user.to_string()] = needed_power_level
+            # Perform a deepcopy of the dict in order to not modify initial_state, as its
+            # contents are preserved as the state for the old room later on
+            initial_state = copy.deepcopy(initial_state)
```

Comment:
Do we really need to copy the entire state rather than just the power levels event?

(it might also be preferable just to copy the bits we know we want from the PL body)

Reply:
Changed to only deepcopy the power levels state. Let me know what you think.